### PR TITLE
detect different ip address

### PR DESCRIPTION
**Pull request by lynchee-owo**
if localhost or ip address, continue
if realchar.ai, add api. at the beginning to be "api.realchar.ai"

### CODE REVIEW

File: client/web/src/hooks/useWebsocket.js [hunk 1]
```diff
@@ -26,6 +27,13 @@ const useWebsocket = (token, onOpen, onMessage, selectedModel) => {
             var ipAddress = parts[0];
             var currentPort = parts[1];
 
+            // Check if the host is 'localhost', IP address or 'realchar.ai'
+            if (ipAddress === 'localhost' || (isIP(ipAddress) || isIPv4(ipAddress))) {
+                // continue with the current host
+            } else if (ipAddress === 'realchar.ai') {
```
Comment (pycui) [inline client/web/src/hooks/useWebsocket.js:33]: Do not hardcode

File: client/web/src/hooks/useWebsocket.js [hunk 2]
```diff
@@ -26,6 +27,13 @@ const useWebsocket = (token, onOpen, onMessage, selectedModel) => {
             var ipAddress = parts[0];
```
Comment (pycui) [inline client/web/src/hooks/useWebsocket.js:27]: Should be called hostname instead of ipadderss

File: client/web/src/components/Auth/SignIn.js [hunk 3]
```diff
@@ -17,12 +18,18 @@ export const sendTokenToServer = async (token) => {
   var parts = currentHost.split(':');
   var ipAddress = parts[0];
   var newPort = '8000';
+  // Check if the host is 'localhost', IP address or 'realchar.ai'
+  if (ipAddress === 'localhost' || (isIP(ipAddress) || isIPv4(ipAddress))) {
+    // continue with the current host
+  } else if (ipAddress === 'realchar.ai') {
```
Comment (pycui) [inline client/web/src/components/Auth/SignIn.js:24]: Ditto

File: client/web/src/hooks/useWebsocket.js (hunk 1 above)
Comment (lynchee-owo) [reply client/web/src/hooks/useWebsocket.js:33]: do you mean do not hardcode realchar.ai? or both realchar.ai and localhost

File: client/web/src/components/Auth/SignIn.js [hunk 4]
```diff
@@ -9,20 +9,27 @@ import React, { useState } from 'react';
...
   // Send token to server
   const scheme = window.location.protocol;
   var currentHost = window.location.host;
   var parts = currentHost.split(':');
-  var ipAddress = parts[0];
+  var hostname = parts[0];
   var newPort = '8000';
-  var newHost = ipAddress + ':' + newPort;
+
+  if (hostname === 'localhost' || (isIP(hostname) || isIPv4(hostname))) {
```
Comment (pycui) [inline client/web/src/components/Auth/SignIn.js:22]: nit: It's best to only avoid having empty condition. e.g. here we can do `if (hostname !== 'localhost' && (isIP(hostname) && isIPv4(hostname))) {...}

**Pull request merged**